### PR TITLE
Delete comments

### PR DESCRIPTION
**Pull request by kassenq**
- Add DeleteDataServlet.java
- Add button to delete comments
- Add JS function with chaining to handle POST request and prevent delay

### CODE REVIEW

File: portfolio/src/main/webapp/comments.html [hunk 1]
```diff
@@ -1,67 +1,65 @@
...
+        <input type="checkbox" name="sort" value="true" /> Alphabetize <br /><br />
+        <button class="btn btn-default" type="submit">Submit</button>
+      </form>
 
-      </div>
-    </body>
-  </html>
+      <br /><br />
+      <p>Pick the number of previous comments:</p>
+      <!-- <input type="number" name="max-num-comments" min="1" max="10" value="1"> -->
```
Comment (qihuang2) [inline portfolio/src/main/webapp/comments.html:49]: nit: remove

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that deletes comment data. */
+@WebServlet("/delete-data")
+public class DeleteDataServlet extends HttpServlet {
+  static final Gson GSON = new Gson();
+  static final String NAME = "name";
```
Comment (qihuang2) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:38]: looks like all these keys are the same as the ones used in DataServlet. let's move them into a new class Keys, so DataServlet and DeleteDataServlet can use the variables

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  static final String TEXT = "text";
+  static final String TIMESTAMP = "timestamp";
+  static final String NAME_INPUT = "name-input";
+  static final String TEXT_INPUT = "text-input";
+  static final String EMPTY_STRING = "";
+  static final String COMMENT_KIND = "Comment";
+  static final String MAX = "max";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (qihuang2) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:48]: does the tutorial say to use doPost? if not, then maybe doDelete is the more suitable method here:  https://docs.oracle.com/javaee/7/api/javax/servlet/http/HttpServlet.html#doDelete-javax.servlet.http.HttpServletRequest-javax.servlet.http.HttpServletResponse-

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -12,40 +12,60 @@
...
   return liElement;
 }
 
-
-
+/**
+ * Creates POST request to delete existing comments.
+ */
+function deleteCommentData() {
+  const request = new Request('/delete-data', {method: POST});
```
Comment (qihuang2) [inline portfolio/src/main/webapp/script.js:69]: if we decide to use doDelete, you might need to update method to DELETE

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 3 above)
Comment (kassenq) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:48]: I think the tutorial says use a POST request

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+public class DeleteDataServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query(Keys.COMMENT_KIND).addSort(Keys.TIMESTAMP, SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      Key commentEntityKey = entity.getKey();
+      datastore.delete(commentEntityKey);
```
Comment (SuperAmin) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:46]: see comment: 
https://github.com/jagjennings/my-portfolio/pull/6#discussion_r436057450

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 3 above)
Comment (qihuang2) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:39]: got it, let'd leave as post then

File: portfolio/src/main/java/com/google/sps/data/Keys.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+// limitations under the License.
+
+package com.google.sps.data;
+import com.google.gson.Gson;
+
+/*
+ * Store Keys used in servlets.
+ */
+public final class Keys {
+  public static final Gson GSON = new Gson();
```
Comment (qihuang2) [inline portfolio/src/main/java/com/google/sps/data/Keys.java:22]: nit: GSON is not really a key. I think keeping GSON in DataServlet.java was fine

**Pull request merged**